visualization/app/codeCharta/ui/copyToClipboardButton/getFilenamesWithHighestMetrics.ts

```diff
@@ -0,0 +1,45 @@
+import { hierarchy } from "d3-hierarchy"
+import { CodeMapNode, NodeType } from "../../codeCharta.model"
+
+export function getFilenamesWithHighestMetrics(fileStates: CodeMapNode) {
+	return fileStates[0].file.map
+}
+
+export function parse(map: CodeMapNode): FileNamesByMetrics {
+	const fileNamesByMetrics = new Map<string, FileToValue[]>()
+	for (const { data } of hierarchy(map)) {
+		if (data.type === NodeType.FILE && data.attributes) {
+			for (const [key, value] of Object.entries(data.attributes)) {
+				maybeAddToMap(key, value, data.name, fileNamesByMetrics)
+			}
+		}
+	}
+	return new Map<string, string[]>()
+}
+
+type FileNamesByMetrics = Map<string, string[]>
+type FileToValue = { name: string; value: number }
+//sry Ruben
+function maybeAddToMap(key: string, value: number, fileName: string, map: Map<string, FileToValue[]>) {
```

Comment:
Please try to find a better name. The method is adding key/value pairs per Metric with highest value to a limit of 10 entries per metric.

Reply:
Resolved by commit https://github.com/MaibornWolff/codecharta/issues/2744